library/core/src/num/f128.rs

```diff
@@ -0,0 +1,1198 @@
+
+    /// Archimedes' constant (π)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const PI: f128 = 3.14159265358979323846264338327950288_f128;
+
+    /// The full circle constant (τ)
+    ///
+    /// Equal to 2π.
+    #[unstable(feature = "f128", issue = "none")]
+    pub const TAU: f128 = 6.28318530717958647692528676655900577_f128;
+
+    /// π/2
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_PI_2: f128 = 1.57079632679489661923132169163975144_f128;
+
+    /// π/3
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_PI_3: f128 = 1.04719755119659774615421446109316763_f128;
+
+    /// π/4
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_PI_4: f128 = 0.785398163397448309615660845819875721_f128;
+
+    /// π/6
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_PI_6: f128 = 0.52359877559829887307710723054658381_f128;
+
+    /// π/8
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_PI_8: f128 = 0.39269908169872415480783042290993786_f128;
+
+    /// 1/π
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_1_PI: f128 = 0.318309886183790671537767526745028724_f128;
+
+    /// 2/π
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_2_PI: f128 = 0.636619772367581343075535053490057448_f128;
+
+    /// 2/sqrt(π)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_2_SQRT_PI: f128 = 1.12837916709551257389615890312154517_f128;
+
+    /// sqrt(2)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const SQRT_2: f128 = 1.41421356237309504880168872420969808_f128;
+
+    /// 1/sqrt(2)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const FRAC_1_SQRT_2: f128 = 0.707106781186547524400844362104849039_f128;
+
+    /// Euler's number (e)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const E: f128 = 2.71828182845904523536028747135266250_f128;
+
+    /// log<sub>2</sub>(10)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const LOG2_10: f128 = 3.32192809488736234787031942948939018_f128;
+
+    /// log<sub>2</sub>(e)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const LOG2_E: f128 = 1.44269504088896340735992468100189214_f128;
+
+    /// log<sub>10</sub>(2)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const LOG10_2: f128 = 0.301029995663981195213738894724493027_f128;
+
+    /// log<sub>10</sub>(e)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const LOG10_E: f128 = 0.434294481903251827651128918916605082_f128;
+
+    /// ln(2)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const LN_2: f128 = 0.693147180559945309417232121458176568_f128;
+
+    /// ln(10)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const LN_10: f128 = 2.30258509299404568401799145468436421_f128;
+}
+
+// TODO: this is probably all wrong
+#[cfg(not(test))]
+impl f128 {
```

Comment:
Is there a library that we pull all these values from, or do we calculate them by hand?